etl/src/main/java/org/pih/hivmigration/etl/sql/VitalsMigrator.groovy

```diff
@@ -0,0 +1,183 @@
+            CREATE TABLE hivmigration_vitals (
+              source_encounter_id int,
+              sign varchar(30),
+              result double,
+              result_unit varchar(100),
+              source_patient_id int,
+              KEY `source_encounter_idx` (`source_encounter_id`)
+            );
+        ''')
+
+        loadFromOracleToMySql('''
+            INSERT INTO hivmigration_vitals (source_encounter_id, sign, result, result_unit, source_patient_id)
+            VALUES (?, ?, ?, ?, ?)
+        ''', '''
+            select 
+                v.ENCOUNTER_ID, 
+                v.SIGN, 
+                v.RESULT, 
+                v.RESULT_UNIT,
+                e.PATIENT_ID
+            from HIV_EXAM_VITAL_SIGNS v, HIV_ENCOUNTERS e
+            where v.ENCOUNTER_ID = e.ENCOUNTER_ID 
+            and e.PATIENT_ID is not null
+            and v.RESULT is not null  -- there are 28 rows with no result
+        ''')
+
+        executeMysql("Create encounter_ids reference table", '''
+            CREATE TABLE hivmigration_vitals_encounters (
+                source_encounter_id int,
+                encounter_id int PRIMARY KEY AUTO_INCREMENT,
+                KEY `source_encounter_idx` (`source_encounter_id`)
+            );
+        ''')
+
+        setAutoIncrement("hivmigration_vitals_encounters", "(SELECT max(encounter_id)+1 FROM encounter)")
+
+        executeMysql("Populate encounter_ids reference table", '''
+            INSERT INTO hivmigration_vitals_encounters (source_encounter_id)
+            SELECT DISTINCT
+                source_encounter_id
+            FROM hivmigration_vitals;
+        ''')
+
+        executeMysql("Create Vitals encounters", '''
+            SET @et = (SELECT encounter_type_id FROM encounter_type WHERE uuid = '4fb47712-34a6-40d2-8ed3-e153abbd25b7');
+            SET @formId = (SELECT form_id from form WHERE uuid = '68728aa6-4985-11e2-8815-657001b58a90');
+            INSERT INTO encounter (encounter_id, encounter_datetime, date_created, encounter_type, form_id, patient_id, creator, location_id, uuid)
+            SELECT
+                ve.encounter_id,
+                e.encounter_date,
+                e.date_created,
+                @et,
+                @formId,
+                p.person_id,
+                COALESCE(hu.user_id, 1),
+                COALESCE(e.location_id, 1),
+                uuid()
+            FROM hivmigration_vitals v
+                JOIN hivmigration_vitals_encounters ve ON v.source_encounter_id = ve.source_encounter_id
+                JOIN hivmigration_encounters e ON v.source_encounter_id = e.source_encounter_id
+                JOIN hivmigration_patients p ON e.source_patient_id = p.source_patient_id
+                JOIN hivmigration_users hu ON e.source_creator_id = hu.source_user_id
+            GROUP BY e.source_encounter_id;
+        ''')
+
+        executeMysql("Create migrate_tmp_obs variant for vitals migration", '''
```

Comment:
Can you explain why we have this stored procedure?  Since it is specialized for this job, and you only ever call it once, wouldn't it be easier and clearer to just execute this insert statement directly at the end of the job?

Reply:
Ha, yep :facepalm: